frontend/src/components/compiler/CompilerOpts.tsx

```diff
@@ -63,17 +64,19 @@ function FlagSet({ name, children, value }) {
 }
 
 function DiffFlagSet({ name, children, value }) {
-    const { setFlag } = useContext(OptsContext)
+    const { setFlags } = useContext(OptsContext)
 
     return <div className={styles.flagSet}>
         <div className={styles.flagSetName}>{name}</div>
         <Select
             onChange={event => {
-                for (const child of children) {
-                    setFlag(child.props.flag, false)
-                }
+                const trueFlag = (event.target as HTMLSelectElement).value
 
-                setFlag((event.target as HTMLSelectElement).value, true)
+                const edits = children.map(child => {
+                    return { flag: child.props.flag, value: child.props.flag == trueFlag }
```

Comment:
I'm hesistant here because `DiffFlagSet`'s `children` arg isn't typed, but I guess pretty much none of this file is typed so its fine?